src/controllers/connectionManager.ts

```diff
@@ -389,7 +389,7 @@ export default class ConnectionManager {
                 Utils.showErrorMsg(Utils.formatString(LocalizedConstants.msgConnectionError, result.errorNumber, result.errorMessage));
                 // check whether it's a firewall rule error
                 let firewallResult = await this.firewallService.handleFirewallRule(result.errorNumber, result.errorMessage);
-                if (firewallResult) {
+                if (firewallResult.result && firewallResult.ipAddress) {
```

Comment:
What result would we get that didn't also have an ip address?

Reply:
false

Reply:
As in we'd get a firewallResult object whose result property was set to false?

Reply:
Yes, that's right